### PR TITLE
HAI-1395 Add attachments page to cable report application form

### PR DESCRIPTION
**Pull request by markohaarni**
# Description

User can add new attachments to application and those are saved when user navigates to next or previous page or saves and quits. Existing attachments are also listed and can be deleted.

## Type of change

- [ ] Bug fix
- [x] New feature
- [ ] Other

# Instructions for testing

Please describe tests how this change or new feature can be tested.

# Checklist:

- [x] I have written new tests (if applicable)
- [x] I have ran the tests myself (if applicable)
- [ ] I have made necessary changes to the documentation, link to confluence
      or other location:

# Other relevant info

Please describe here if there is e.g. some requirements for this change or
other info that the tester/user needs to know.

### CODE REVIEW

File: src/domain/application/attatchments.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import api from '../api/api';
+import { ApplicationAttachmentMetadata, AttachmentType } from './types/application';
+
+// Get attachments metadata related to an application
+export async function getAttachments(applicationId: number | null | undefined) {
```
Comment (pitkni) [inline src/domain/application/attatchments.ts:5]: There's a typo in this files name.

File: src/domain/johtoselvitys/Attachments.tsx [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+}: Props) {
+  const queryClient = useQueryClient();
+  const { t } = useTranslation();
+  const locale = useLocale();
+  const { getValues } = useFormContext<JohtoselvitysFormValues>();
+
+  const [fileToDelete, setFileToDelete] = useState<ApplicationAttachmentMetadata | null>(null);
+  const [showFileDeleteDialog, setShowFileDeleteDialog] = useState(false);
+  const attachmentDeleteMutation = useMutation(deleteAttachment, {
+    onSuccess() {
```
Comment (pitkni) [inline src/domain/johtoselvitys/Attachments.tsx:37]: How does this work if the application is already deleted or it is not found (404)?

File: src/domain/application/attatchments.ts (hunk 1 above)
Comment (markohaarni) [reply src/domain/application/attatchments.ts]: Fixed.

File: src/domain/johtoselvitys/Attachments.tsx (hunk 2 above)
Comment (markohaarni) [reply src/domain/johtoselvitys/Attachments.tsx:37]: Do you mean application or attachment? If the application is not found, this component is not even rendered so there should be no problem.

Comment (pitkni) [reply src/domain/johtoselvitys/Attachments.tsx:37]: Sorry, I meant attachment.

**Review by pitkni — APPROVED** [review]

**Pull request merged**